forestmvey/aws-secrets-manager-rotation-lambdas
# Add InfluxDB Multi User Token rotation Lambda

*Issue #, if available:*
N/A

*Description of changes:*
The InfluxDB Multi User Token Rotation Lambda rotates a users authentication token using the AWS Secrets Manager. The master-user rotation scheme utilizes a master token that does not change, and a user token that is periodically rotated. For InfluxDB this requires that the secret manager token(master) must be an operator token. The benefit of using this scheme is that the user can have any permission set, rather than needing the permissions required to create and rotate the token.

InfluxDB does not support creating authentication token with custom values. This means that the lambda function updates the secret values with those generated by token creation in InfluxDB. This removes the need for the `setSecret` step since the stage is already set, and the token value is created in the `createSecret` step.

*Verifications done*
- [x] Deployed lambda, created secret, and verified token rotation.  

*How to test*
- Add `requests` to `requirements.txt`
- At the top of `LambdaFunction:LambdaHandler` add the following lines:
```
ip = get('https://api.ipify.org').content.decode('utf8')
print('My public IP address is: {}'.format(ip))
```
- After you first attempt a rotation you can look in the cloudwatch logs to find the IP address.
- Add this IP address to the inbound traffic rules for the InfluxDB Instance you are testing rotation with.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
